starter/app/src/main/java/com/udacity/project4/locationreminders/reminderslist/ReminderListFragment.kt

```diff
@@ -47,41 +89,108 @@ class ReminderListFragment : BaseFragment() {
 
     override fun onResume() {
         super.onResume()
-        //load the reminders list on the ui
+        Log.d(TAG, "ReminderListFragment.onResume().")
+
+        logoutClicked = false
+        // Load the reminders list on the ui
         _viewModel.loadReminders()
     }
 
+    //--------------------------------------------------
+    // Main Methods
+    //--------------------------------------------------
+
+    private fun init() {
+        Log.d(TAG, "ReminderListFragment.init().")
+        parent = requireActivity()
+        parent.onBackPressedDispatcher.addCallback(viewLifecycleOwner, backPressHandler)
+
+        val menuHost: MenuHost = parent
+        addMenu(menuHost)
+
+        setDisplayHomeAsUpEnabled(false)
+        setTitle(getString(R.string.app_name))
+
+        binding.refreshLayout.setOnRefreshListener { _viewModel.loadReminders() }
+
+        requestPermissionListener()
+    }
+
     private fun navigateToAddReminder() {
-        //use the navigationCommand live data to navigate between the fragments
+        Log.d(TAG, "ReminderListFragment.navigateToAddReminder().")
+        // Use the navigationCommand live data to navigate between the fragments
         _viewModel.navigationCommand.postValue(
             NavigationCommand.To(
-                ReminderListFragmentDirections.toSaveReminder()
+                ReminderListFragmentDirections.actionReminderListFragmentToSaveReminderFragment()
             )
         )
     }
 
     private fun setupRecyclerView() {
-        val adapter = RemindersListAdapter {
-        }
-
-//        setup the recycler view using the extension function
+        Log.d(TAG, "ReminderListFragment.setupRecyclerView().")
+        val adapter = RemindersListAdapter {}
+        // Setup the recycler view using the extension function
         binding.reminderssRecyclerView.setup(adapter)
     }
 
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        when (item.itemId) {
-            R.id.logout -> {
-//                TODO: add the logout implementation
```

Comment:
@graffiti75 Where I can find tis TODO in your updated code?

Reply:
New PR created: https://github.com/udacity/nd940-android-kotlin-c4-starter/pull/14